src/modules/0.0.22/aws_s3/mappers/bucket_object.ts

```diff
@@ -34,10 +34,11 @@ export class BucketObjectMapper extends MapperBase<BucketObject> {
     return res;
   };
 
-  getBucketObjects = paginateBuilder<S3>(paginateListObjectsV2, 'Contents', undefined, undefined, (b, k) => ({
-    Bucket: b,
-    Prefix: k,
-  }));
+  getBucketObjects = paginateBuilder<S3>(paginateListObjectsV2, 'Contents', undefined, undefined, (b, k) => {
+    const out: any = { Bucket: b };
+    if (k) out.Prefix = k;
```

Comment:
Just curious what `k` was here. Currently the `out.Prefix` would be `undefined`. So it was something like `''` or `false`? :D

Reply:
There's unfortunately some *weirdness* with Javascript and `undefined`.

![Screenshot from 2022-11-09 17-34-46](https://user-images.githubusercontent.com/765551/200964779-3aea5a87-7b9d-4eb7-9976-33d158eaa8c1.png)

Most of the time, this distinction doesn't matter, but tools that do meta-work on the data structures themselves, like the underlying JSON-to-XML stuff that the AWS SDK does, may end up producing weird XML output like casting an `undefined` into an empty string, which could have follow-on results.

I made that change out of an abundance of caution. Not sure if it actually was also a secondary cause of listing failures or not, but I don't see a reason to undo it, personally. ;)

Reply:
Got it. Thanks for that tip.